File: X42.scala
object X42 {
  def main(args: Array[String]) = {
    if (args.length > 0 && args(0) == "42") {
      System.err.println("It depends!");
      System.exit(42);
    }
    System.out.println("What is the meaning of life?");
  }
}

File: X42.go
package main

import "os"

func main() {
	if len(os.Args) > 1 && os.Args[1] == "42" {
		os.Stderr.WriteString("It depends!\n")
		os.Exit(42)
	}
	os.Stdout.WriteString("What is the meaning of life?\n")
}

